src/opnsense/mvc/app/controllers/OPNsense/Dnsmasq/Api/SettingsController.php

```diff
@@ -262,4 +371,54 @@ public function delBootAction($uuid)
     {
         return $this->delBase('dhcp_boot', $uuid);
     }
+
+    /**
+     * Return selectpicker options for interfaces and tags
+     */
+    public function getTagListAction()
+    {
+        $result = [];
+
+        // Interfaces
+        $interfaces = [];
+        foreach (Config::getInstance()->object()->interfaces->children() as $key => $intf) {
+            if ((string)$intf->type === 'group') {
+                continue;
+            }
+
+            $interfaces[] = [
+                'value' => $key,
+                'label' => empty($intf->descr) ? strtoupper($key) : (string)$intf->descr
+            ];
+        }
+        usort($interfaces, fn($a, $b) => strcasecmp($a['label'], $b['label']));
+
+        // Tags
+        $tags = [];
+        foreach ($this->getModel()->dhcp_tags->iterateItems() as $tag) {
+            $name = trim((string)$tag->tag);
+            if (!empty($name)) {
+                $tags[] = [
+                    'value' => $name,
+                    'label' => $name
+                ];
+            }
+        }
+        usort($tags, fn($a, $b) => strcasecmp($a['label'], $b['label']));
+
+        // Assemble result
+        return [
+            'tags' => [
+                'label' => gettext('Tags'),
+                'icon'  => 'fa fa-tag text-primary',
```

Comment:
Although the icons are funny, if we do want to use them more often we might want to consider adding a wrapper for the pattern. (see also further below)